test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,337 @@
+	}
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+	addToFrameworkSchemeForSmokeTests(t)
+	suite.deployJaegerWithPropagationEnabled()
+}
+
+func (suite *TokenTestSuite) TearDownSuite() {
+	undeployJaegerInstance(suite.exampleJaeger)
+	handleSuiteTearDown()
+}
+
+func (suite *TokenTestSuite) TestTokenPropagationNoToken() {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
+		resp, err := client.Do(req)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusForbidden, resp.StatusCode)
+		return true, nil
+	})
+	require.NoError(t, err, "Token propagation test failed")
+}
+
+func (suite *TokenTestSuite) TestTokenPropagationValidToken() {
+	/* Create an span */
+	collectorPort := randomPortNumber()
+	collectorPorts := []string{collectorPort + ":14268"}
+	portForwColl, closeChanColl := CreatePortForward(namespace, suite.collectorName, collectorPodImageName, collectorPorts, fw.KubeConfig)
+	defer portForwColl.Close()
+	defer close(closeChanColl)
+	collectorEndpoint := fmt.Sprintf("http://localhost:%s/api/traces", collectorPort)
+	cfg := config.Configuration{
+		Reporter:    &config.ReporterConfig{CollectorEndpoint: collectorEndpoint},
+		Sampler:     &config.SamplerConfig{Type: "const", Param: 1},
+		ServiceName: testServiceName,
+	}
+	tracer, closer, err := cfg.NewTracer()
+	require.NoError(t, err, "Failed to create tracer in token propagation test")
+	tStr := time.Now().Format(time.RFC3339Nano)
+	tracer.StartSpan("TokenTest").
+		SetTag("time-RFC3339Nano", tStr).
+		Finish()
+	closer.Close()
+	client := newHTTPSClient()
+	/* Try to reach query endpoint */
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, suite.queryServiceEndPoint, nil)
+		req.Header.Add("Authorization", "Bearer "+suite.token)
+		resp, err := client.Do(req)
+		defer resp.Body.Close()
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+		if resp.StatusCode != http.StatusOK {
+			return false, errors.New("Query service returns http code: " + string(resp.StatusCode))
+		}
```

Comment:
The if is redundant here as if the require.Equal fails the test will terminate.  You should pick one or the other.  In general I prefer the require as it makes for more concise code and produces better error messages.  However, if you prefer the if you need to change the string(resp.StatusCode) to strconv.Itoa

Reply:
I preferred required too, but I had to do in this way, so if it fails the first time due the pod not fully ready or other timing issues, the wait.Poll will retry again.

If I use required, it will fail mediately. Making this test less reliable. IMHO.

I removed the required line.